src/context/CustomerChargeContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext } from "react";
+import useCustomerCharge from "../hooks/useCustomerCharge";
+import { CustomerCharge } from "../type";
+
+interface CustomerChargeContextInterface {
+  customerCharge: CustomerCharge;
+  addCustomerCharge: (chargeToAdd: number) => void;
+  subtractCustomerCharge: (chargeToSubtract: number) => void;
+  returnAllCustomerCharge: () => void;
+  resetReturnedChange: () => void;
+}
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const CustomerChargeContext =
+  createContext<CustomerChargeContextInterface | null>(null);
+
+export const CustomerChargeProvider = ({ children }: Props) => {
```

Comment:
저는 children props 은 따로 타입핑 안해주고, `PropsWithChildren`을 사용하는데! 어떠신가요??
지금은 props 로 받아오는 값이 children 밖에 없어서 `unknown` 으로 해주었지만 나중에 Props 가 추가되면 저 자리에 interface 를 넣어주면 되더라구요! 근데 이건 취향차이라 ㅎㅎ 
```suggestion
export const CustomerChargeProvider = ({ children }: React.PropsWithChildren<unknown>) => {
```

Reply:
와앙 요런 아이가 있었군요! 
`React.PropsWithChildren<{}>` 이렇게 적용해주면 unknown을 사용하지 않고도 쓸 수 있네요!
배워갑니다! ><

https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/context/#extended-example